packages/pg-protocol/src/buffer-writer.ts

```diff
@@ -5,7 +5,7 @@ export class Writer {
   private offset: number = 5
   private headerPosition: number = 0
   constructor(private size = 256) {
-    this.buffer = Buffer.alloc(size)
+    this.buffer = Buffer.allocUnsafe(size)
```

Comment:
Curious -- is changing to `allocUnsafe` here also contributes to performance improvements? The original PR comment doesn't mention anything about this change.
Also how certain we are that this doesn't cause any nasty side-effects of being, well, unsafe?

Reply:
Here's a link to the [NodeJS documentation](https://nodejs.org/docs/latest-v12.x/api/buffer.html#buffer_static_method_buffer_allocunsafe_size) on `Buffer.allocUnsafe()`.

I'm not an expert on Buffers, though the documentation's last sentenance says this can increase a performance (subtle) over `Buffer.alloc()`.

> Allocates a new Buffer of size bytes. If size is larger than buffer.constants.MAX_LENGTH or smaller than 0, ERR_INVALID_OPT_VALUE is thrown.
>
> The underlying memory for Buffer instances created in this way is not initialized. The contents of the newly created Buffer are unknown and may contain sensitive data. Use Buffer.alloc() instead to initialize Buffer instances with zeroes.
>
> [...]
>
> Use of this pre-allocated internal memory pool is a key difference between calling `Buffer.alloc(size, fill)` vs. `Buffer.allocUnsafe(size).fill(fill)`. Specifically, `Buffer.alloc(size, fill)` will never use the internal Buffer pool, while `Buffer.allocUnsafe(size).fill(fill)` will use the internal Buffer pool if size is less than or equal to half `Buffer.poolSize`. The difference is subtle but can be important when an application requires the additional performance that Buffer.allocUnsafe() provides.